server/src/main/java/org/opensearch/search/SearchBootstrapSettings.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Keeps track of all the search related node level settings which can be accessed via static methods
+ *
+ * @opensearch.internal
+ */
+public class SearchBootstrapSettings {
+    // settings to configure maximum slice created per search request using OS custom slice computation mechanism. Default lucene
+    // mechanism will not be used if this setting is set with value > 0
+    public static final String CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_KEY = "search.concurrent.max_slice";
+    public static final int CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_DEFAULT_VALUE = -1;
```

Comment:
Hm ... the values <= 0 are not valid, how come we have -1 as the default? Shouldn't we align with Apache Lucene default here (MAX_SEGMENTS_PER_SLICE) and prohibit the usage of invalid values?

Reply:
So values <=0 is used to fallback to lucene mechanism to compute the slices whereas > 0 is used for the new mechanism introduced in this PR.

Reply:
> So values <=0 is used to fallback to lucene mechanism

This is implementation choice, the issue is - what it means from the user perspective, and I would argue that negative values are not useful. If one wants to set the value of this setting, it should be meaningful, and Apache Lucene has a sensible default for it.

Reply:
Out of all options, I would suggest to have pseudo `auto` value, to fallback to default implementation (yes, it will change the type of the setting) or, at last, 0.

Reply:
This setting is controlling 2 things: a) Whether to use custom slice logic b) If yes, then what target slice count is. Negative values from user perspective means to disable this custom slice computation logic and +ve values are for setting the target slice value. 

There is no relation with the lucene defaults here as the mechanism is different than what is provided by lucene. We are not using this setting to control default lucene behavior.

Reply:
Just realized that, thanks @sohami , let use `0` as the fallback to default, the negative values should not be allowed for this setting I think.